src/main/java/com/google/sps/data/FirestoreDB.java

```diff
@@ -64,58 +64,60 @@ public void deleteUser(String userId) {
       DBUtils.blockOnFuture(DBUtils.user(userId).delete());
     }
     
-    /** Add recipe id: true as a singleton map to user's created recipes field. */
-    public void addRecipeIdToCreated(String userId, String recipeId) {
+    /**
+     * Sets a property's value to "true" for a certain user document.
+     * Can be used to let a user add a recipe to saved or created, or to let user follow a tag.
+     * @param userId the user's Firebase ID
+     * @param objectId the ID of either a recipe if the intent is to save/create, or of a tag for tag following.
+     * @param collection a constant from User class indicating which mode -- save, create, or follow tag.
+     */
+    public void makeUserPropertyTrue(String userId, String objectId, String collection) {
       DocumentReference userRef = DBUtils.user(userId);
-      String nestedPropertyName = DBUtils.getNestedPropertyName(User.CREATED_RECIPES_KEY, recipeId);
-      ApiFuture addToCreatedPostsFuture =
-          userRef.update(Collections.singletonMap(nestedPropertyName, true));
-      DBUtils.blockOnFuture(addToCreatedPostsFuture);
-    }
-    
-    /** Remove a recipe from a user's created (when recipe is deleted). */
-    public void removeRecipeIdFromCreated(String userId, String recipeId) {
-      DocumentReference userRef = DBUtils.user(userId);
-      String nestedPropertyName = DBUtils.getNestedPropertyName(User.CREATED_RECIPES_KEY, recipeId);
-      Map<String, Object> update = new HashMap<>();
-      update.put(nestedPropertyName, FieldValue.delete());
-      DBUtils.blockOnFuture(userRef.update(update));
-    }
-    
-    /** Add recipe id: true as a singleton map to user's saved recipes field. */
-    public void addRecipeIdToSaved(String userId, String recipeId) {
-      DocumentReference userRef = DBUtils.user(userId);
-      String nestedPropertyName = DBUtils.getNestedPropertyName(User.SAVED_RECIPES_KEY, recipeId);
-      ApiFuture addToSavedPostsFuture =
+      String nestedPropertyName;
+      
+      switch (collection) {
```

Comment:
Again, the switch statement here is unnecessary. Ideally, the `makeUserPropertyTrue` should be called with param `collection` as one of `User.CREATED_RECIPES_KEY, User.SAVED_RECIPES_KEY, User.TAGS_FOLLOWED_KEY`.

Reply:
ah good point